_posts/2025-05-15-express-cleanup-legacy-packages.md

```diff
@@ -0,0 +1,46 @@
+---
+title: "Spring Cleaning in Express.js: Deprecations and the Path Ahead"
+tags: news
+author: Express Technical Committee
+description: As part of a broader effort to modernize and streamline Express.js, we’ve deprecated several outdated packages including csurf, connect-multiparty, and path-match. Learn why we made these changes and what it means for the future of the framework.
+---
+
+As Express.js continues to power web applications across the world, it's important that we maintain a clean, reliable, and modern codebase. In that spirit, we've done a bit of spring cleaning.
+
+Over the past few weeks, we've evaluated legacy packages within the Express.js ecosystem—some of which have become outdated, unmaintained, or misaligned with modern best practices. As a result, we’ve officially deprecated several of them.
+
+## 🚨 Packages Deprecated
+
+Here are the key packages we’ve deprecated:
+
+- [`csurf`](https://www.npmjs.com/package/csurf): A CSRF middleware that’s long been difficult to maintain and is better handled today through frameworks or custom implementations that align with your architecture.
+- [`connect-multiparty`](https://www.npmjs.com/package/connect-multiparty): A multipart form-data parser that relies on deprecated libraries and hasn’t aged well.
+- [`path-match`](https://www.npmjs.com/package/path-match): A route-matching utility that has been superseded by more modern and maintained alternatives.
+
+Each of these packages was originally created to solve real problems—but time has moved on, and the ecosystem has evolved.
+
+## 🤔 Why This Matters
+
+Maintaining deprecated or inactive dependencies introduces technical debt and security risk. By formally deprecating these packages, we:
+
+- Encourage developers to adopt better-maintained and more secure solutions.
+- Reduce confusion around which tools are actively supported by Express.
+- Focus our efforts on modernizing the core and surrounding ecosystem.
+
+## 🗂️ What You Should Do
+
+If your application depends on any of these packages, now is a great time to look for alternatives. For instance:
+
+- Consider modern CSRF protection strategies tied to your specific backend or framework.
```

Comment:
I feel like folks will read this line and ask "but my framework is express? what should I use with express?".  I know the idea is to not take an opinion, but maybe we could link to the [npm search page for other csrf libraries](https://www.npmjs.com/search?q=csurf)? I think the best replacement one is in that list at least (`csrf-csrf`)? Because the other option is our recommendation is "migrate to fastify, they have this correctly implemented", which is also valid but I doubt what folks come to the express blog to hear.

Reply:
I think that this solves the issue a059d9a. I prefer not recommending a library unless we are clear on that "project recommendation".